### PR TITLE
Avoid a redirect loop if we have a login error

### PR DESCRIPTION
**Pull request by Calvario**
Please give a short description what your pull request is for

Avoid a redirect loop if we have a toasts (login error)

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Calvario** [comment]
I understand your point, the issue is that sometimes the login "error" can be on our side and not on the SSO side.
It wouldn't be inside the request, exemple with this try/catch:
- https://github.com/librenms/librenms/blob/master/app/Http/Controllers/Auth/SocialiteController.php#L124

So as it's working right now, the toast is the correct thing to check.

**Comment by murrant** [comment]
@Calvario thanks for the code link... I see the correct fix there.  After, the toast, you should return with a page reload instead of letting it fall through to the redirect.

**Comment by Calvario** [comment]
Moved to draft as discussed in Discord, I will then change both SocialiteController.php & LoginController.php to read an option to stop the redirect in case of error.

File: app/Http/Controllers/Auth/SocialiteController.php [hunk 1]
```diff
@@ -122,6 +122,8 @@ private function login(string $provider): RedirectResponse
             return redirect()->intended();
         } catch (AuthenticationException $e) {
             toast()->error($e->getMessage());
+
+            return redirect()->route('login')->with('block_auto_redirect', true);
```
Comment (murrant) [inline app/Http/Controllers/Auth/SocialiteController.php:126]: Yeah, I messed this up a bit..  you need to delete one of these two return statements (either location is the same)

**Comment by Calvario** [comment]
@murrant ready for review, I tested it and it stopped correctly the auto-redirect.

**Review by murrant — APPROVED** [review]

**Pull request merged**